### PR TITLE
fix: version for tags with `-arabica`, `-mocha`, `-rc` suffixes

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3933

## Testing 

```
$ git checkout v2.2.0-mocha
$ make debug-version
GIT_TAG: v2.2.0-mocha
VERSION: 2.2.0-mocha
$ git tag --delete v2.2.0-mocha
Deleted tag 'v2.2.0-mocha' (was d3f8b2997)
$ make debug-version
GIT_TAG: v2.2.0-arabica
VERSION: 2.2.0-arabica
$ git tag --delete v2.2.0-arabica
Deleted tag 'v2.2.0-arabica' (was d3f8b2997)
$ make debug-version
GIT_TAG: v2.2.0-rc0
VERSION: 2.2.0-rc0
$ git tag --delete v2.2.0-rc0
Deleted tag 'v2.2.0-rc0' (was d3f8b2997)
$ make debug-version
GIT_TAG: v2.1.2-4-gd3f8b2997
VERSION: 2.1.2-4-gd3f8b2997
```

### CODE REVIEW

**Review by ninabarbakadze — APPROVED** [review]

**Review by rach-id — APPROVED** [review]
utACK

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+# not have metadata like creatordate. Therefore, this command is a hacky attempt
+# to get the most recent tag on the current commit according to Celestia's
+# testnet versioning scheme + SemVer.
+GIT_TAG := $(shell git tag --points-at HEAD --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$$' \
+    || git tag --points-at HEAD --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+-mocha$$' \
+    || git tag --points-at HEAD --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+-arabica$$' \
+    || git tag --points-at HEAD --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+-rc[0-9]*$$' \
+    || git tag --points-at HEAD --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+-(beta)$$' \
+    || git tag --points-at HEAD --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+-(alpha)$$' \
+    || git describe --tags)
```
[the remark this answers is not in the document]
Comment (cmwaters) [reply Makefile:16]: Maybe you want to do this ^

**Review by cmwaters — APPROVED** [review]
Thanks for the fix. It seems like the regex could be consolidated

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (rootulp) [reply Makefile:16]: The proposed command won't use the priority we want (v2.2.0, v2.2.0-mocha, v2.2.0-alpha, etc)

**Pull request merged**